src/hooks/redis.js

```diff
@@ -0,0 +1,51 @@
+import { safeRequire } from '../utils/requireUtils';
+import { hook } from '../extender';
+import { getRandomId } from '../utils';
+import { createRedisSpan, extendRedisSpan } from '../spans/redisSpan';
+import { SpansContainer } from '../globals';
+import * as logger from '../logger';
+
+const createCallbackHandler = redisSpan => args => {
+  const ended = new Date();
+  const [error, result] = args;
+  const span = extendRedisSpan(redisSpan, {
+    ended,
+    error,
+    result,
+  });
+  SpansContainer.addSpan(span);
+};
+
+function sendCommandBeforeHook(args) {
+  const command = args[0];
+  if (this.ready === false || this.stream.writable === false || command.command === 'info') {
```

Comment:
How do we have `this.read` and `this.stream`? Does `this.stream` always defined?

Reply:
Fixed the `this.stream`